fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,185 @@
+
+            while (dataFileReader.hasNext()) {
+                avroData = dataFileReader.next(avroData);
+
+                // Creates a JsonObject from the AVRO GenericRecord object
+                JsonNode avroDataJson = MAPPER.readTree(new JsonFactory().createParser(avroData.toString()));
+
+                // Constructs a Map of the converted dimension field name and the corresponding value in the avro file
+                Map<String, String> dimensionFieldKeyValues = dimension.getDimensionFields()
+                        .stream()
+                        .collect(Collectors.toMap(dimensionField -> dimensionField.getName(),
+                                                  // Looks up converted field name from the `convertedDimensionFields`
+                                                  // Parsed as text to avoid the extra quotes around string objects
+                                                  dimensionField -> avroDataJson
+                                                        .get(convertedDimensionFields
+                                                                .get(dimensionField.getName())
+                                                        ).asText()
+                                                )
+                        );
+
+                // Parses the dimensionRow based on the map generated above and then adds it to the dimension cache
+                dimension.addDimensionRow(dimension.parseDimensionRow(dimensionFieldKeyValues));
+            }
+        }
+    }
+
+    /**
+     * Constructs a JSON Node object from the avro schema file.
+     *
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return JsonNode object
+     *
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    private JsonNode convertFileToJsonNode(String avroSchemaPath) throws IOException {
+        File file;
+        JsonNode jsonNode;
+        try {
+            file = new File(avroSchemaPath);
+            jsonNode = MAPPER.readTree(file);
+            return jsonNode;
+        } catch (JsonProcessingException e) {
+            LOG.error("Unable to process the Json ", e.getStackTrace());
+            throw e;
+        } catch (IOException e) {
+            LOG.error("Unable to process the file ", e.getStackTrace());
+            throw e;
+        }
+    }
+
+    /**
+     * Returns a converter function for the dimension field name mapping.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param dimensionFieldMapper Object that defines the dimension field name transformations
+     *
+     * @return Function that converts the dimension field name based on the user mapping
+     */
+    private Function<DimensionField, String> curry(Dimension dimension, DimensionFieldMapper dimensionFieldMapper) {
+        return dimensionField -> dimensionFieldMapper.converter(dimension, dimensionField);
```

Comment:
The action of this method is to curry, but the meaning is to map the fieldNames of dimension fields for a given dimension.  The name would be better if it corresponded with the meaning of the method.